src/query/api/v1/handler/database/kvstore_test.go

```diff
@@ -0,0 +1,186 @@
+// Copyright (c) 2021 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package database
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+
+	"github.com/m3db/m3/src/cluster/generated/proto/kvpb"
+	"github.com/m3db/m3/src/cluster/kv"
+	"github.com/m3db/m3/src/dbnode/kvconfig"
+)
+
+func TestUpdateQueryLimits(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	tests := []struct {
+		name          string
+		limits        *kvpb.QueryLimits
+		commit        bool
+		expectedJSON  string
+		expectedError string
+	}{
+		{
+			name:         `nil`,
+			limits:       nil,
+			commit:       true,
+			expectedJSON: "",
+		},
+		{
+			name:         `empty`,
+			limits:       &kvpb.QueryLimits{},
+			commit:       true,
+			expectedJSON: "",
+		},
+		{
+			name: `only block - commit`,
+			limits: &kvpb.QueryLimits{
+				MaxRecentlyQueriedSeriesBlocks: &kvpb.QueryLimit{
+					Limit:           1,
+					LookbackSeconds: 15,
+					ForceExceeded:   true,
+				},
+			},
+			commit:       true,
+			expectedJSON: `maxRecentlyQueriedSeriesBlocks:<limit:1 lookbackSeconds:15 forceExceeded:true > `,
```

Comment:
out of curiosity what's up with the formatting at the end (`true > `)

Reply:
Oh so the `true` is the val of `forceExceeded` and the `>` is the close bracket for the `json.RawMessage`

Reply:
(was more curious why the spacing ends up like that instead of the beginning, which doesn't have any spaces, not really important)